packages/cmf/BREAKING-CHANGES.md

```diff
@@ -0,0 +1,84 @@
+# 1.0 -> 2.0
+
+## extract out react-router (#1842)
+
+[PR #1842](https://github.com/Talend/ui/pull/1842)
+[jira TUI-271](https://jira.talendforge.org/browse/TUI-271)
+
+First setup it reading the [@talend/react-cmf-router documentation](https://www.npmjs.com/package/@talend/react-cmf-router).
+
+### matchPath
+
+```javascript
+//before
+import matchPath from '@talend/react-cmf/lib/sagaRouter/matchPath';
+//after
+import matchPath from '@talend/react-cmf/lib/matchPath';
+```
+
+### selectors
+
+```javascript
+//before
+import cmf from '@talend/react-cmf';
+cmf.selectors.router.getPath(state);
+cmf.selectors.router.getLocation(state);
+//after
+import { routerAPI } from '@talend/react-router';
+routerAPI.selectors.getPath(state);
+routerAPI.selectors.getLocation(state);
+```
+
+### context
+
+```javascript
+// before
+function myActionCreator(event, data, context) {
+    if (context.router.params.myParamId) {
+        return {
+            type: 'MY_ACTION_WITH_PARAM',
+            id: context.router.params.myParamId
+        };
+    }
+    return {
+        type: 'MY_ACTION_WITHOUT_ROUTER',
+    };
+}
+// after
+function myActionCreator(event, data, context) {
+    const match = routerAPI.selectors.matchPath(context.store.getState(), {
+        path: '/your/path/:myParamId', // you need that extra
+    });
+    if (match) {
+        return {
+            type: 'MY_ACTION_WITH_PARAM',
+            id: match.params.myParamId
+        };
+    }
+    return {
+        type: 'MY_ACTION_WITHOUT_ROUTER',
+    };
+}
+```
+
+Why do I need to write down a path here ?
+Because before you rely on the router configuration, so you may without know it
+adding a route that will match this and provide a `myParamId`. This is not enough explicit, your code should not rely on configuration.
```

Comment:
can we agree to remove the whole cmfRouter thing then ? this abstraction was created so we could leverage settigns to create route and stuff.